cfpb/consumerfinance.gov
# Convert is_archived field to a choice field with never archive option

We will soon be implementing an automatic nightly archiving script to archive anything that was published more than two years ago. As we may want exceptions to this rule, we needed a way to tell this script not to archive certain pages.

This commit converts `is_archived` from a boolean field into a `CharField` with defined choices, one of which is `'never'`, which, if chosen, will tell the forthcoming script to not archive that page.

## Changes

- `is_archived` field on `CFGOVPage` changed to multiple-choice `CharField` with new option to never automatically archive a page
- Adds `archived` property to CFGOVPage to return a boolean interpretation of the new yes/no/never options

## Removals

- Removes `filter_archived` option from the `FilterableList` block, as it is now obsolete


## How to test this PR

1. Pull branch
1. Migrate
1. Set an item's `is_archived` field to Yes
1. Confirm that the filters still work as expected

_N.B.:_ The "Never" option will not have any visible effect and will only be used when the automatic archiving script is updated to take it into account. 


## Notes and todos

- [x] Push commit to remove obsolete `filter_archived` field on the block


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
